src/components/Swap.tsx

```diff
@@ -66,13 +66,26 @@ export const Swap = (): JSX.Element => {
 	const [fromValue, setFromValue] = useState('');
 	const [toValue, setToValue] = useState('');
 	const { data: tokensData } = useTokens();
-
 	const [fromToken, setFromToken] = useState<Token | undefined>();
 	const [fromAmount, setFromAmount] = useState<string>();
 	const [toToken, setToToken] = useState<Token>();
 	const [toAmount, setToAmount] = useState<string>();
```

Comment:
We should store the amounts separately as `BigNumbers`. I've found this pretty convenient, mostly to avoid rounding error when parsing from and  to `string`.

Something like this 
```type InputAmount = {value: BigNumber, displayValue: string }```.

You can use the display value for the value in the input component and transform it to big number on change.

Reply:
In which case would we lose precision by going from BigNumber to string or viceversa? The problem comes if we use something like `toNumber`, where we allow JS to truncate numbers based on its maximum precision; but we are not, nor should we, doing that afaik

Reply:
Let's YAGN for now.